Allow composition of previously generated components.

tnez opened the pull request:
I think this could be useful for building up more complicated components out of smaller. Kind of like a static version of the composition we see in react / riot components.

tnez committed:
Allow composition of previously generated components.

I think this could be useful for building up more complicated components
out of smaller. Kind of like a static version of what we see in react /
riot components.

burlesona commented:
I like this idea. I'm not sure how I feel about using `compose` as the instruction, though, since it feels to me like that implies more functionality than is really present. It's not doing anything special to "compose" elements, your method just includes raw HTML. That's potentially really useful though.

So with that in mind, how do you feel about the method name being `raw` instead? That way it's more obvious that you could do inline stuff as well, something like:

``` ruby
html do
  raw "<!-- I'm a comment that will show up in HTML! -->"
end
```

tnez committed:
Change compose to raw per @burlesona.

I agree, I think `raw` is a better representation of what is going
on. Good call @burlesona.

tnez commented:
I'm down with that. I think you're right - it's a better representation of what's going on. Made the change.

burlesona commented:
Hey, sorry I spaced and didn't merge this yet. Taking care of it now :)
